homeassistant/components/automation/__init__.py

```diff
@@ -787,6 +788,7 @@ def _trigger_extract_device(trigger_conf: dict) -> list[str]:
         trigger_conf[CONF_PLATFORM] == "event"
         and CONF_EVENT_DATA in trigger_conf
         and CONF_DEVICE_ID in trigger_conf[CONF_EVENT_DATA]
+        and isinstance(trigger_conf[CONF_EVENT_DATA][CONF_DEVICE_ID], str)
     ):
         return [trigger_conf[CONF_EVENT_DATA][CONF_DEVICE_ID]]
```

Comment:
If it's a list, shouldn't we also return the list of device IDs?

Reply:
It cannot be a list :)
As in, an event cannot originate from multiple entities at the same time.